cunumeric/module.py

```diff
@@ -3032,7 +3154,8 @@ def _contract(
     else:
         c_dtype = ndarray.find_common_type(a, b)
     a = _maybe_cast_input(a, c_dtype, casting)
-    b = _maybe_cast_input(b, c_dtype, casting)
+    if b is not None:
+        b = _maybe_cast_input(b, c_dtype, casting)
```

Comment:
I changed `_maybe_cast_input` to not handle a None input. That greatly simplified its type signature at the expense of this conditional.